feat: polls should be able to end a gamble but leave the poll open

streeeggs opened the pull request:
New button to end betting without closing the poll. Also prevent guests from voting/betting

<img width="1886" height="304" alt="image" src="https://github.com/user-attachments/assets/98bcc62e-970b-4860-a63f-bd6c39a1aeee" />
<img width="1908" height="309" alt="image" src="https://github.com/user-attachments/assets/cc6be94f-36bc-41c3-9d0f-8537955c2ad3" />
left: admin; right: guest

streeeggs committed:
feat: polls should be able to end a gamble but leave the poll open

streeeggs committed:
fix: cleanup

streeeggs committed:
fix: broke stuff

esnyder232 committed:
fixed winning ratio bug. Added server side check on closed gamble poll. Added more logging.

esnyder232 commented on src/channel/coolhole-poll.js:
```diff
@@ -274,6 +275,14 @@ CoolholePollModule.prototype.handleVote = function (user, data) {
+        message: `Poll is no longer open.`,
+      });
+      return;
+    }
```
Added server side check for if the poll is closed.

esnyder232 commented on src/channel/coolhole-poll.js:
```diff
@@ -289,6 +298,11 @@ CoolholePollModule.prototype.handleVote = function (user, data) {
+      if (this.poll.gamble) {
+          this.channel.logger.log("[poll] For gambling poll '" + this.poll.title + "', user " + user.getName() + " gambled " + data.wager + " points on option " + parseInt(data.option));
+      }
+
```
Additional logging for debugging in prod.

esnyder232 commented on src/channel/coolholepoints.js:
```diff
@@ -924,7 +924,7 @@ class Coolpoints extends ChannelModule {
           );
           const shareOfThePot =
-            Math.round(wager / totalWagersOfWinners) * totalWagers;
+            Math.round(wager / totalWagersOfWinners * totalWagers);
```
Fixed winning ratio bug.
People were saying they weren't getting winnings even though they picked the winning option.
The math.round was rounding incorrectly; ie. people who bet "100" in a winning pot of "1000" would round the ratio to 0 instead of 0.1. 
Changed the rounding so it rounds the points instead of the ratio.

esnyder232 committed:
Fixed error message popup.

streeeggs committed:
fix: gambleStatus to show proper button on load and after ending bet

streeeggs commented:
@esnyder232 All good changes. Changed a few minor things and fixed `End Betting` button to switch to `End Poll` when clicked. Should also work after load as well; noticed that it wasn't saved to channel data properly either

streeeggs committed:
fix: end poll button replacement and options should grow when winning option button is gone

streeeggs committed:
fix: only show winning options when gamble is closed
